mapper/dm_agent/src/batcher.rs

```diff
@@ -121,9 +121,14 @@ impl MessageBatcher {
         let collectd_message = CollectdMessage::parse_from(&first_message)?;
         let mut message_batch =
             MessageBatch::start_batch(collectd_message, first_message_timestamp)?;
+        let sleep = time::sleep(self.batching_window);
```

Comment:
A code comment that the "real sleep" doesn't happen here but rather inside the select loop might help devs who read this code in future.

Reply:
I agree that a comment is required. However, the proposed comment is misleading.

Line 124, starts a timer but doesn't wait for its completion. The `sleep` variable a `Future` handler that is awaited line 129. Hence, we are collecting messages at most `batching_window` seconds.